src/pages/index.js

```diff
@@ -143,29 +240,80 @@ function getCardElement(data) {
   return cardElement;
 }
 
-initialCards.forEach((item) => {
-  const cardElement = getCardElement(item);
-  cardsList.append(cardElement);
-});
-
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalInputName.value;
-  profileDescription.textContent = editModalInputDescription.value;
-  closeModal(editModalProf);
+  const editModalSaveBtn = evt.submitter;
+  setButtonText(editModalSaveBtn, true);
+
+  api
+    .editUserInfo({
+      name: editModalInputName.value,
+      about: editModalInputDescription.value,
+    })
+    .then((userData) => {
+      profileName.textContent = userData.name;
+      profileDescription.textContent = userData.about;
+      closeModal(editModalProf);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      setButtonText(editModalSaveBtn, false);
+    });
 }
 
 function handleNewCardSubmit(evt) {
   evt.preventDefault();
-  const inputValues = { link: cardLinkInput.value, name: cardNameInput.value };
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-  evt.target.reset();
-  disableButton(cardSubmitBtn, settings);
+  const cardSubmitBtn = evt.submitter;
+
+  setButtonText(cardSubmitBtn, true);
+  api
+    .addNewCard({ name: cardNameInput.value, link: cardLinkInput.value })
+    .then((cardData) => {
+      const inputValues = {
+        link: cardData.link,
+        name: cardData.name,
+      };
+      const cardElement = getCardElement(inputValues);
+      cardsList.prepend(cardElement);
+      closeModal(newCardModal);
+      evt.target.reset();
+      disableButton(cardSubmitBtn, settings);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      setButtonText(cardSubmitBtn, false);
+    });
+}
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const avatarSubmitBtn = evt.submitter;
 
-  closeModal(newCardModal);
+  setButtonText(avatarSubmitBtn, true);
+  avatarSubmitBtn.disabled = true;
+
+  api
+    .avatarSubmit(avatarInput.value)
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+      evt.target.reset();
+      disableButton(avatarSubmitBtn, settings); //not sure
+      closeModal(profileAvatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      avatarSubmitBtn.disabled = false;
```

Comment:
`avatarSubmitBtn.disabled = false;` should be only in `.catch` otherwise you **enable** the button, though you want to disable it in line 304


```js
    })
    .catch((err) => {
      console.error(err);
      avatarSubmitBtn.disabled = false;  // here
    })
    .finally(() => {
      setButtonText(avatarSubmitBtn, false);
    });
```